rewrite-core/src/main/java/org/openrewrite/marker/RecipesThatMadeChanges.java

```diff
@@ -32,4 +34,70 @@ public class RecipesThatMadeChanges implements Marker {
     UUID id;
 
     Collection<Stack<Recipe>> recipes;
+
+    /**
+     * A list of recipe names and their associated options which have made changes to the AST
+     * @param indent string prepended to sub recipes for logging a hierarchical view
+     */
+    @Incubating(since = "7.22.0")
+    public List<String> recipeNamesAndOptionsThatMadeChanges(String indent) {
+        List<RecipeDescriptor> recipesToDisplay = new ArrayList<>();
+
+        for (Stack<Recipe> currentStack : recipes) {
+
+            Recipe root = currentStack.get(1);
+            RecipeDescriptor rootDescriptor = root.getDescriptor().withRecipeList(new ArrayList<>());
+
+            RecipeDescriptor index;
+            if (!recipesToDisplay.contains(rootDescriptor)) {
+                recipesToDisplay.add(rootDescriptor);
+                index = rootDescriptor;
+            } else {
+                index = recipesToDisplay.get(recipesToDisplay.indexOf(rootDescriptor));
+            }
+
+            for (int i = 1; i < currentStack.size(); i++) {
+                Recipe child = currentStack.get(i);
+                if (child == root) {
```

Comment:
It looks like this condition will always continue on the first iteration.
```java
Recipe root = currentStack.get(1);
```

and

```java
            for (int i = 1; i < currentStack.size(); i++) {
                Recipe child = currentStack.get(i);
```

Reply:
yeah agreed, you could just set i=2 and remove the continue altogether.

Reply:
yeah.  thanks!